### ISSUE TITLE
Not connecting to Bitbucket API

### ISSUE THREAD

**Issue by swedishboy** [afragen/github-updater#659]
version 7.5.0

message:

"This repository has not connected to the API or was unable to connect."

anyway of getting a more verbose log/output telling me what's wrong?

**Comment by afragen** [comment]
This is likely due to the new WP-Cron background processing. I’ve changed the message in the `develop` branch to show that it’s waiting on WP-Cron. The error should clear within a minute or so as long as WP-Cron runs.

**Comment by swedishboy** [comment]
we've disabled wp-cron (on the site with the error) and uses crontab instead. is this affecting the github-updater? wp-cron.php is called every 5 minutes from crontab.

the problem we have is that we can't update our private rep. theme or plugins anymore. no other branches can be selected so it doesn't seemt to connect to bitbucket.

**Comment by afragen** [comment]
It means that GitHub Updater will only check every 5 minutes with the cache in place for 12 hours. You can manually clear the cache using the _Refresh Cache_ button.

If you use the following filter you can bypass wp-cron.
`add_filter( 'github_updater_disable_wpcron', '__return_true' );`

Installing Query Monitor can assist with any API debugging.

**Comment by swedishboy** [comment]
thank you. disabling wpcron for github made the connections work again.
maybe add this as an configurable option for those who disabled wp-cron and using crontab?

**Comment by afragen** [comment]
Did you see the wiki page on [Background Processing](https://github.com/afragen/github-updater/wiki/Background-Processing)?

You can use crontab, you simply need the proper command to run the wp-cron.php and/or the waiting wp-cron events.

**Comment by swedishboy** [comment]
the cronjob actually seemed to work. but we lost the connection to bitbucket. if i went into themes and checked branch switching nothing came up.

but we're investigating that our cronjobs work correctly with nginx. i'll get back to you if that was the issue that stopped the connection to bitbucket.

**Comment by swedishboy** [comment]
i also noticed before using the filter to turn cron off for github-updater that the events fired used very long strings of data.

**Comment by afragen** [comment]
Can you provide an example? Query Montior is a great tool for debugging this stuff.

**Comment by swedishboy** [comment]
we're using the plugin WP Crontrol to see what cronjobs are scheduled in WP.
your cronjob list the following arguments:

`[
    {
        "github-updater": {
            "type": "github_plugin",
            "uri": "https://github.com/afragen/github-updater",
            "enterprise": null,
            "enterprise_api": null,
            "owner": "afragen",
            "repo": "github-updater",
            "branch": "master",
            "slug": "github-updater/github-updater.php",
            "local_path": "/mnt/persist/www/docroot/plugins/github-updater/",
            "extended_repo": "github-afragen-github-updater",
            "author": "Andy Fragen",
            "name": "GitHub Updater",
            "local_version": "7.5.0",
            "sections": {
                "description": "
This plugin was designed to simply update any GitHub hosted WordPress plugin or theme. Currently, plugins or themes hosted on GitHub, Bitbucket, or GitLab are also supported. Additionally, self-hosted installations of GitHub or GitLab are supported. It also allows for remote installation of plugins or themes into WordPress.

\n
Your plugin or theme must contain a header in the style.css header or in the plugin's header denoting the location on GitHub. The format is as follows.

\n
GitHub Plugin URI: afragen/github-updater

\n
GitHub Plugin URI: https://github.com/afragen/github-updater

\n
or

\n
GitHub Theme URI: afragen/test-child

\n
GitHub Theme URI: https://github.com/afragen/test-child

\n
...where the above URI leads to the owner/repository of your theme or plugin. The URI may be in the format https://github.com/<owner>/<repo> or the short format <owner>/<repo>. You do not need both. Only one Plugin or Theme URI is required. You must not include any extensions like .git.

\n
The following headers are available for use depending upon your hosting source.

\n
GitHub

\n
\n
GitHub Plugin URI
\n
GitHub Theme URI
\n
GitHub Languages
\n
\n
Bitbucket

\n
\n
Bitbucket Plugin URI
\n
Bitbucket Theme URI
\n
Bitbucket Languages
\n
\n
GitLab

\n
\n
GitLab Plugin URI
\n
GitLab Theme URI
\n
GitLab Languages
\n
GitLab CI Job
\n
\n
Comprehensive information regarding GitHub Updater is available on the wiki.

\n
Slack

\n
We now have a Slack team for GitHub Updater. Please click here for an invite. You will be automatically added to the #general and #random channels. Please take a look at other channels too.

",
                "changelog": "
[unreleased]

\n
7.5.0 / 2018-01-28

\n
\n
fixed View detail ratings for large projects with lots of issues
\n
fixed API::set_readme_info() to see passed parameter as readme data
\n
added title attribute to icons on Settings subtabs, thanks @petemolinero
\n
created new class Init to help unclutter class Base
\n
fixed PHP Warning if saving empty Remote Management Settings
\n
changed some variable and function names to be more descriptive
\n
moved Singleton Factory out of namespace
\n
moved capabilities check into class Init
\n
moved API classes to subdirectory
\n
moved WP-CLI classes to subdirectory
\n
refactored autoloader to grab all subdirectories
\n
fixed for new WP.org Plugin API response
\n
updated vendor/class-parser.php and vendor/persist-admin-notices-dismissal
\n
fixed composer.json for new license format
\n
\n
7.4.4 / 2017-11-29

\n
\n
fixed bug in remote install where Bitbucket credentials weren't transferred to Basic_Auth_Loader, #630
\n
\n
7.4.3 / 2017-11-07

\n
\n
set all extra header values in Base::parse_extra_headers()
\n
added more error messaging for class WP_Error
\n
fixed some issues with GitHub Release Assets
\n
\n
7.4.2 / 2017-10-25

\n
\n
added check to see if wp-cron is updating and if not send and error message
\n
fix for WP-CLI updating for private Bitbucket repos, thanks @v8-ict
\n
\n
7.4.1 / 2017-10-22

\n
\n
oops, during refactor of Install I copied the incorrect query for GitHub's remote install
\n
\n
7.4.0 / 2017-10-21

\n
\n
use wp-cron for background processing of wp_remote_get() calls for getting repo data \ud83d\ude80
\n
fixed #603 by not creating generic global variables accidentally
\n
fixed issue with remote install of private Bitbucket repos
\n
added plugin icons to update-core.php page for WP 4.9
\n
fixed stale AWS download link for GitHub release asset
\n
cache get_plugins() and wp_get_themes() for short period giving better performance to some admin pages, fixes #612
\n
refactor of methods from class Base to class API
\n
created class API_PseudoTrait to share methods of class API, workaround for OOP traits
\n
fixed removal of stale options
\n
\n
7.3.1 / 2017-09-20

\n
\n
removed parent constructor from Branch, thanks @fwolfst
\n
\n
7.3.0 / 2017-09-15

\n
\n
removed non-constructor stuff from all constructors
\n
added parent::__construct() to extended classes where needed
\n
fixed #568, thanks @bradmkjr
\n
fixed multisite bug for theme update rows that I introduced in v7.0.0 :-(
\n
fixed PHP notice #591
\n
fixed bug with current branch data being deleted when saving settings with refactor of Settings::filter_options()
\n
fixed issues with up to date notice during branch switch #598
\n
\n
7.2.0 / 2017-08-30

\n
\n
added a static proxy class to use for creating Singletons
\n
fixed Override Dot Org for themes
\n
fixed PHP Notice #584
\n
fixed bug introduced in readme.txt parsing #589
\n
fixed bug introduced in v7.0.0 with linter updates to properly display multisite theme updates in themes.php
\n
fixed branch setting bug #592 by moving trigger from filter hook to direct call, thanks @rob and @idpaterson
\n
\n
7.1.0 / 2017-08-10

\n
\n
always show Install button for single site theme when branch switch is active #567
\n
fixed override of dot org to correctly ignore dot org updates #581
\n
no more extended naming
\n
added constant for overriding dot org updates when plugins have identical slugs, GITHUB_UPDATER_OVERRIDE_DOT_ORG replacing the GITHUB_UPDATER_EXTENDED_NAMING constant
\n
added Overriding Dot Org functions for both plugins and themes
\n
\n
7.0.0 / 2017-08-01

\n
\n
added support for GitLab Groups #556, thanks @rolandsaven
\n
refactored Settings and Install to place API Settings data in individual API classes
\n
refactored Settings to make smaller methods
\n
simplified composer.json, removed autoload section and no need to require composer/installer
\n
many PHP Inspections fixes
\n
fixed class Rest_Update for PHP 5.3 compatibility, thanks @epicfaace
\n
created class Branch to automatically set correct branch during branch switch or install. No more need for Branch header. This is a breaking change as master will become the default branch for all repositories. You will need to use Branch Switch to reinstall the current branch for it to be correctly set.
\n
\n
6.3.5 / 2017-06-29

\n
\n
hotfix to composer.json to remove classmap and files, I think I messed something up.
\n
\n
6.3.4 / 2017-05-28

\n
\n
fixed #547 for RESTful updating after breaking it again
\n
fixed PHP errors #550
\n
\n
6.3.3 / 2017-05-16

\n
\n
definitive fix for #549
\n
update to class-parser.php@5483
\n
\n
6.3.2 / 2017-05-09

\n
\n
added broken setting to repo not returning HTTP 200 for the main file
\n
~fixed PHP error #549~
\n
added div class to Settings page to create more specific CSS selectors
\n
\n
6.3.1 / 2017-05-01

\n
\n
simplify uninstall.php
\n
ensure Basic Auth headers are loaded for RESTful updating #547
\n
\n
6.3.0 / 2017-04-26

\n
\n
fixed to not run load_pre_filters() during WP-CLI, fixes #528 thanks @egifford
\n
hopefully fixed annoying, intermittent PHP notices empty parse_header_uri() output
\n
added a singleton to class Settings to avoid duplicate loads #531
\n
refactored subtabs for Settings page
\n
refactored parsing of extra headers, Enterprise and CE headers no longer needed
\n
added support for Bitbucket Server!! Thanks @lkistenkas for access and especially to @BjornW for kicking it off
\n
refactored add_endpoints() to use everywhere
\n
now requires WordPress 4.4 and above
\n
update to latest wp.org class-parser.php
\n
move enqueuing of plugin CSS to Base::init()
\n
refactored Language Pack updating to their own classes
\n
split out abstract methods from abstract class API to interface API_Interface
\n
make Autoloader better functioning as a drop-in
\n
switched logic for plugin branch switching and setting the update transient
\n
refactor add_access_token_endpoint() to class API
\n
refactor Basic Authentication headers to class Basic_Auth_Loader
\n
moved checkboxes before titles in Settings
\n
updated wiki screenshots
\n
fixed to call load_options() in Base::init() to properly utilize options
\n
add red (#f00) warning dashicon in Settings for repo with malformed header URI
\n
\n
6.2.2 / 2017-02-09

\n
\n
fixed for updating via webhook from GitHub tagged release, declare branch as master
\n
refactored Install download link generation
\n
fixed PHP notices #525
\n
replaced method with mb_strrpos() in class-parser.php as some users don't have this function
\n
fixed JSON syntax error in GitHub webhook payload
\n
fixed GitLab Install tab to always show access token
\n
fixed GitLab Settings to show individual access tokens
\n
\n
6.2.1 / 2017-02-02

\n
\n
removed wp_cache_flush() for Install page, not needed with Base::admin_pages_update_transients() 
\n
hotfix for upgrade routine to properly flush caches :P
\n
\n
6.2.0 / 2017-02-02

\n
\n
added WP-CLI compatibility
\n
refactored Base::admin_pages_update_transient() and API::wp_update_response() to use Base::make_update_transient_current(), this fixed some PHP notices #508
\n
added banner display to plugin View details iframe
\n
change API::get_dot_org_data to use JSON response to avoid PHP notices
\n
refactored GitHub_API::get_repo_meta() for simplification
\n
moved some repo renaming to their own methods from Base::upgrader_source_selection() to Base::fix_misnamed_directory(), Base::extended_naming(), and Base::fix_gitlab_release_asset_directory()
\n
moved a couple class-parser.php mods to separate functions in class Readme_Parser
\n
refactored GitHub_API::get_repo_meta() to use more efficient API call, gets forks also, thanks @egifford
\n
introduce some variability to transient expiration per plugin
\n
switch to storing repo data in options table instead of using transients, this should help with object caching which doesn't like transients
\n
fixed branch switching with extended naming #520, thanks @joelworsham
\n
updated continuous integration via RESTful endpoints to also update based upon a new tag/release of the repo
\n
\n
6.1.1 / 2016-11-29

\n
\n
hotfix to flush cache during upgrade routine
\n
\n
6.1.0 / 2016-11-28

\n
\n
improved transient saving to save optimized version of transient rather that whole API response
\n
changed Refresh Cache to POST to only run once.
\n
fixed API::wp_update_response to properly reset the update transient after a shiny update or cache flush
\n
added Base::admin_pages_update_transient to properly reset the update transient on plugins.php and themes.php pages
\n
fixed Bitbucket authentication during AJAX update
\n
changed to use dashicon to identify private repos in Settings
\n
fixed transient update when doing shiny updates
\n
added ability to update from GitHub release asset
\n
added our own PHP version check
\n
refactored setting of update transient during rollback, should eliminate the up to date message and rollback failures
\n
added class GHU_Upgrade to run upgrade functions if needed
\n
fixed initial display of update for dot org plugins with higher version numbers on git repos when they should be updating from dot org 496
\n
refactored query to wp.org for plugin data
\n
revert javascript href call because Firefox can't have nice things
\n
fixed to allow themes to rollback at any time
\n
renamed filter hook github_updater_token_distribution to github_updater_set_options as more descriptive
\n
added deprecated hook notice for github_updater_token_distribution
\n
fixed setting of GitLab meta
\n
changed to not skip setting meta when no update available
\n
fixed uninstall.php for option not transient
\n
\n
6.0.0 / 2016-10-26

\n
\n
added class Language_Pack and new repo, Language Pack Maker, to create and update from a separate Language Pack repository.
\n
added new header for Language Pack updates. Language Pack updates can and will now be decoupled from the plugin release.
\n
obfuscated token/password values in Settings page, for @scarstens
\n
added support for GitLab Build Artifacts as Release Assets, #459
\n
improved check for private repo, removes public repos from Settings page when no updates are available
\n
improved to provide Settings page with dynamically displayed sub-tabs
\n
added display of installed plugins/themes using GitHub Updater in Settings sub-tabs
\n
added ability to enter Bitbucket credentials to Install tabs if not already present
\n
moved action/filter hook calls out of constructors, make @carlalexander happy
\n
improved to incorporate GitLab personal access tokens, users will need to reset tokens.
\n
added a filter hook 'github_updater_run_at_scale' to skip several API calls making GitHub Updater at scale more performant, see README for usage details
\n
added several hooks for  WP REST Cache and @scarstens
\n
skip API calls for branches and tags if branch switching not enabled
\n
refactored delete_all_transients() to delete from database, only called in class Base
\n
refactored and improved branch switching to be consistent among plugins and themes. This means plugins now can rollback to one of the previous 3 tagged releases.
\n
fixed get_repo_slugs() for initially misnamed repository, ie github-updater-develop
\n
renamed Refresh Transients to Refresh Cache, hopefully to provide more clarity
\n
refactored to only load GHU site options and other database queries for privileged users on backend only
\n
added query arg of ?per_page=100 to GitLab query for project IDs, this is max number able to be retrieved, yes an edge case #465
\n
\n
5.6.2 / 2016-09-24

\n
\n
added reset of update_plugins and update_themes transient with Refresh Transients
\n
throw Exception for webhook update if PUSH is to branch different from webhook
\n
removed translations from RESTful endpoint responses, only visible from webhook or direct call
\n
fixed PHP fatal during heartbeat for class PAnD not found, early exit in class too early, #453
\n
fixed PHP notice in Bitbucket_API, #451
\n
\n
5.6.1 / 2016-09-15

\n
\n
fixed PHP notices when parsing readme.txt with missing data
\n
fixed PHP fatal by namespacing class WordPressdotorg\\Plugin_Directory\\Readme\\Parser
\n
fixed PHP fatal in WordPressdotorg\\Plugin_Directory\\Readme\\Parser by avoiding dereferenced array call
\n
\n
5.6.0 / 2016-09-14

\n
\n
added Refresh Transients button to Settings page because the Check Again button is going away
\n
added redirect_on_save() for Settings page
\n
switched to slightly modified version of wp.org plugin readme parser, now accepts Markdownified readme.txt files
\n
fixed re-activation of RESTful plugin update, multisite vs single site
\n
when creating Settings page, check current Plugin/Theme class instance, not transient. Fixes issue where remote install of private repo not having private settings saved.
\n
fixed PHP errors in Settings page
\n
fixed saving issues with checkboxes during remote install of private Bitbucket repo
\n
added one day dismissal of admin notices using persist-admin-notices-dismissal library
\n
Settings page now uses same function to update settings for both single/multisite
\n
temporary fix for AJAX updates of private Bitbucket repos #432, can only do one per page load, not very AJAXy :P
\n
fixed class Rest_Update to avoid potential race conditions when RESTful endpoint is used as a webhook
\n
added branch and branches to update transient, might be able to use this in RESTful update sometime
\n
fixed extended naming when installing forks of plugins and plugins
\n
\n
5.5.0 / 2016-07-02

\n
\n
better internationalization for changing plugin View details link
\n
refactored and improved class Additions for GitHub Updater Additions plugin
\n
fixed using GitLab CE private token with using class Install
\n
reworked GitHub repo meta as search was occasionally flaky, now also using owner's repos check
\n
refactored adding extra headers
\n
added RESTful endpoints for updating from CLI or browser, courtesy of @limikael
\n
added reset of RESTful API key
\n
added CSS file to help display theme view details
\n
refactored get_remote_{plugin|theme}_meta() to get_remote_repo_meta() as it was in 4 different places :P
\n
updated for Shiny Updates
\n
fixed PHP fatal, thanks @charli-polo
\n
fixed displaying WP_Errors
\n
made error messages non-static
\n
fixed pesky PHP notice when updating from 5.4.1.3 #403
\n
added aria-labels for screen readers
\n
always display theme rollback/branch switcher in single site installation #411
\n
fixed extended naming issue when branch switching, #429
\n
\n
5.4.1 / 2016-04-21

\n
\n
get tags for themes to rollback even if no updates are available. I was overzealous in cutting remote API calls.
\n
ManageWP now works for Remote Management.
\n
fixed bug in GitLab_API to use path and not name. Thanks @marbetschar
\n
added filter for background updates if set globally. Thanks @jancbeck
\n
fixed PHP notice when adding new Remote Management option
\n
deleted all transients on uninstall
\n
fixed logic for display of GitLab token fields and error notice
\n
displayed WP_Error message for wp_remote_get() error
\n
correctly get use GitLab namespace/project instead of project id when needed
\n
added data-slug to theme update rows so CSS may be applied
\n
now supports MainWP for remote management, thanks @ruben-
\n
typecast readme.txt response to array, fix for occasional malformed readme.txt file
\n
\n
5.4.0 / 2016-3-18

\n
\n
fixed deprecated PHP4 constructor in vendor class.
\n
added class Additions to process JSON config from hook to add repos to GitHub Updater, see GitHub Updater Additions
\n
added necessary code in class Plugin and class Theme for above
\n
skipped many remote API calls if no update available and use local files, huge performance boost :-)
\n
removed check for GitHub asset, this eliminates an API call for a rarely used feature
\n
added additional header Release Asset: true to add back ability to set download link to release asset.
\n
added function to remove Basic Authentication header when downloading private Bitbucket release assets as they are stored on AmazonS3 and use Query String Request Authentication Alternative
\n
consolidated error messages to show only once per error
\n
added Other Notes section to View details
\n
updated readme.txt with Other Notes information
\n
\n
5.3.4 / 2016-01-24

\n
\n
reset 'new_version' in update transient to avoid up to date failure with branch switching.
\n
fixed display of branch switching themes on single install.
\n
fixed bug in getting Bitbucket branch names.
\n
fixed to hide checkbox when active as mu-plugin.
\n
work better with shiny updates.
\n
\n
5.3.3 / 2016-01-04

\n
\n
removed added filters, below as they didn't add functionality to this plugin.
\n
try to use references to &$this
\n
added PHPUnit testing setup, I could use help writing tests. A great way to contribute. :-)
\n
\n
5.3.2 / 2015-12-21

\n
\n
code simplification for upgrader_source_selection
\n
fixed plugin branch switching to override up-to-date message (most of the time)
\n
added filters for developers, well I wanted them anyway ;-)\n
\n
github_updater_plugin_transient_update
\n
github_updater_theme_transient_update
\n
github_updater_plugin_row_meta
\n
github_updater_theme_row_meta
\n
github_updater_append_theme_action
\n
\n
fixed renaming of updating plugins that were never initially renamed when first installed. Strange bug.
\n
\n
5.3.1 / 2015-12-03

\n
\n
fixed PHP notice during remote installation
\n
fixed remote install #325
\n
\n
5.3.0 / 2015-11-25

\n
\n
fixed parsing of readme.txt for donate link
\n
refactored transient storage resulting in significantly few database calls, more performant.
\n
moved {get|set}_transient functions to abstract class API
\n
fixed settings page saving errors.
\n
fixed shiny updates #321
\n
overhauled of renaming code back to using upgrader_source_selection and for WordPress 4.4 adding $args['hook_extra'] toupgrader_source_selection` filter. Thanks @dd32!
\n
\n
5.2.0 / 2015-10-14

\n
\n
fixed #309 for proper GitHub Enterprise endpoints
\n
added setting for GitHub Enterprise personal access token
\n
new function _add_access_token() for class GitHub_API
\n
updatede erusev/parsedown to current release
\n
\n
5.1.2 / 2015-09-25

\n
\n
added upgrader_source_selection filter back for correct updating of current, active theme.
\n
fixed #293 and #297
\n
removed pre_http_request filter blocking
\n
fixed javascript for theme rollback - @scarstens
\n
play nice with current master branch of wp-update-php
\n
\n
5.1.1 / 2015-09-09

\n
\n
hotfix to comment out pre_http_request filter. Updating of plugin doesn't work. I need to re-think this one.
\n
\n
5.1.0 / 2015-09-09

\n
\n
refactored Plugin and Theme constructors moving code calling APIs getting remote data to separate functions
\n
fixed #281, removed 'Activate Plugin/Theme' buttons post-install
\n
fixed #284 for GitLab CE/Enterprise install and update
\n
fixed to re-activate plugins after update, doesn't work with branch switching :person_frowning:
\n
fixed to correctly rename plugin/theme on update if installed from upload.
\n
added filter to pre_http_response to bypass certain plugins check using wp_remote_get with each page load in GitHub Updater. Bypass is only for 12 hours.
\n
cosmetic fix to display GitHub Updater as active when activated as mu-plugin
\n
fixed to theme_api 'View version details' CSS; better scrolling for changelog info
\n
fixed annoying PHP notice in vendor/parse-readme.php when Upgrade Notice malformed
\n
fixed API::return_repo_type to add 'type' to array; allows easier instance creation of API classes
\n
updated POT file
\n
\n
5.0.1 / 2015-08-18

\n
\n
updated to current erusev/parsedown release, fixes PHP7 issue
\n
updated to current WPupdatePHP/wp-update-php/release-1-1-0 branch
\n
\n
5.0.0 / 2015-08-15

\n
\n
fix rollback for GitLab themes
\n
add branch switcher for themes
\n
escape all printed strings
\n
changed from using upgrader_source_selection hook to upgrader_post_install, this greatly simplifies renaming
\n
removed class Remote_Update as it's no longer needed when using upgrader_post_install hook
\n
added Remote Management settings tab more cleanly support those services that currently integrate with GitHub Updater
\n
modified the process loading so faster for admin level users. Much thanks @khromov
\n
added hooks for devs to set GitHub Access Tokens and hide the Settings page. Please be sure your client will never need access to the Settings page. Thanks @oncecoupled
\n
fixed #267 thanks @stevehenty and @rocketgenius
\n
\n
4.6.2

\n
\n
refactor remote update services to new class Remote_Update
\n
general security fixes, don't call files directly...
\n
fix/test for remote updating via InfiniteWP. Child themes are not identified by IWP as needing updates, otherwise it seems to work as expected.
\n
\n
4.6.1

\n
\n
fix for remote updating via iThemes Sync
\n
fix for renaming when AJAX updating of plugins
\n
\n
4.6.0

\n
\n
newer, much more precise method for renaming based upon selected repos from the dashboard. Yes, I tested on staging server. :-)
\n
added feature to use extended naming of plugin directories to avoid potential conflict with WP.org slugs. Props @reinink for the idea.
\n
strip .git from the end of the plugin or theme URI for those who haven't gotten to the README yet.
\n
added javascript show/hide options on the Install page.
\n
fixed boolean logic to not display GitLab Private Token input on Install if it's already set.
\n
updated screenshots in README
\n
switched a number of methods to be non-static, anticipation of testing.
\n
broken: renaming during updates from upgrade services
\n
\n
4.5.7

\n
\n
hotfix GitLab private updating/installing
\n
fix some PHP notices
\n
\n
4.5.6

\n
\n
bugfix for renaming code to properly strip <owner>-
\n
most of Russian translation by Anatoly Yumashev
\n
\n
4.5.5

\n
\n
back to simplifying the renaming code, always remember to test renaming on live server.
\n
strip <owner>- and -<hash> from beginning and end of update for more precise renaming
\n
I think this is the end of renaming for a while. :P
\n
\n
4.5.4

\n
\n
hotfix for renaming, I reverted back a bunch with more extensive testing on server. It's amazing how different renaming is locally vs on server.
\n
\n
4.5.3

\n
\n
updated language files -- oops
\n
\n
4.5.2

\n
\n
cleanup and refactor of renaming code.
\n
added Romanian translation by Corneliu Cirlan
\n
added Japanese translation by ishihara
\n
\n
4.5.1

\n
\n
fix bug so updates display without having to randomly refresh.
\n
\n
4.5.0

\n
\n
fix some PHP notices
\n
add update by GitHub release asset in lieu of update by tag when asset is present
\n
install asset via remote install if asset URI used
\n
refactor to simplify class structure, created abstract class API and class Messages
\n
add GitLab support!!
\n
refactor to set all git servers and extra headers to static arrays in Base
\n
remove checkbox when loaded as mu-plugin, props @pbearne
\n
\n
4.4.0

\n
\n
only add custom user agent once :P
\n
add support of GitHub Enterprise via new GitHub Enterprise header
\n
sanitize filter input
\n
add support for parsing readme.txt for View details information using WordPress_Plugin_Readme_Parser by @markjaquith
\n
fixed View details link to show for all cases when plugin using GitHub Updater
\n
refactor creation of header parts and URIs
\n
\n
4.3.1

\n
\n
Spanish translation by Jose Miguel Bejarano
\n
German translation by Linus Metzler
\n
squish PHP notices
\n
add custom user agent to wp_remote_get and tweak error message at request of GitHub ;-)
\n
fixed edge case renaming bug
\n
\n
4.3.0

\n
\n
use @WPUpdatePhp class WPUpdatePhp for PHP version checking
\n
use https://api.wordpress.org not http
\n
Arabic translation by Hyyan Abo FAkher
\n
make strings better for translation - thanks @pedro-mendonca and @fxbenard
\n
additional Portuguese translation by Pedro Mendon\u00e7a
\n
refactor for getting local plugin and theme meta, now simpler for additional APIs (I'm thinking about you GitLab)
\n
fix link in README to GitHub Link
\n
correctly pass array as last argument in add_settings_field
\n
add focus to URI input field
\n
add Setting for personal GitHub Access Token to avoid API rate limit - thanks @mlteal
\n
add Setting for branch switching from the Plugins page
\n
add 'View details' link in Plugins page
\n
\n
4.2.2

\n
\n
fix POT and some updated languages, thanks @fxbenard
\n
fix PHP notice for $options settings on initial install - thanks @benosman
\n
\n
4.2.1

\n
\n
add PHP version check for graceful exit
\n
add to error message for 401 error.
\n
save settings when remote installing a private repo
\n
\n
4.2.0

\n
\n
added minutes until reset of GitHub API's rate limit to error message
\n
added placeholder = \"master\" to remote install branch text input
\n
I should have made the last version 4.2.0 as I added a new feature. I'll try to be better with semantic versioning in the future. ;-)
\n
\n
4.1.4

\n
\n
add message to certain admin pages when API returns HTTP error code
\n
update POT to remove HTML entity codes from strings and generally try to make i18n better
\n
Swedish translation by Andr\u00e9as Lundgren
\n
added logo to README and Settings page
\n
\n
4.1.3

\n
\n
use strtolower comparison of plugin directory and repo name. This might is an issue related to the manual installation of a plugin before any update might occur. This allows the View details screen to display in these instances where the case of the directory and repo aren't identical. This doesn't work for themes.
\n
\n
4.1.2

\n
\n
hide star ratings from View details screen for private repos
\n
\n
4.1.1

\n
\n
add plugin to $response in Plugin::pre_set_site_transient_update_plugins to fix PHP Notice
\n
rename classes to src to follow more conventional naming
\n
refactor renaming code to function under all circumstances, I hope ;-)
\n
\n
4.1.0

\n
\n
added remote installation of plugins or themes, both public and private
\n
remote installation using either full URI or short <owner><repo> format
\n
created new tabbed interface for settings
\n
added another screenshot to readme
\n
I'd like to apologize to all my translators for adding new strings often, you guys are great, thanks!
\n
\n
4.0.1

\n
\n
hotfix to force an array type when sanitizing settings, it gave me a fatal I wasn't expecting.
\n
\n
4.0.0

\n
\n
changed is_a() to instanceof per https://core.trac.wordpress.org/changeset/31188
\n
requires PHP 5.3 or greater as autoloader class requires namespacing\u2028
\n
updated all classes for namespacing
\n
renamed directory and class names to allow for PSR 4 style loading\u2028
\n
clean up a number of foreach loops where I was only using either key or value, not both
\n
Special thanks for all my translators, especially @grappler for adding translation key for description
\n
bugfix to correctly pick CHANGES.MD or CHANGELOG.MD regardless of case
\n
removed reading/saving GitHub Access Token header into settings. Must use Settings Page.
\n
\n
3.2.3 - 3.2.6

\n
\n
added French translation by @daniel-menard
\n
added Italian translation by @overclokk
\n
added Portuguese translation by @valeriosouza
\n
added Ukrainian translation by @andriiryzhkov (our first translation!!)
\n
\n
3.2.2

\n
\n
remove scraping of user/pass from Bitbucket URI as it's no longer needed
\n
use Requires WP header to fill view options detail
\n
rename private methods to begin with underscore
\n
add screenshot to README for Settings Page (only 70 kB)
\n
stop re-creating transient of transients if it already exists
\n
\n
3.2.1

\n
\n
refactored adding extra headers to class GitHub_Updater to ensure they're added before they're needed, resolves issue with WooThemes Updater plugin
\n
update .pot file
\n
\n
3.2.0

\n
\n
changed settings page and how Bitbucket Private repos authenticate with your username/password
\n
update .pot
\n
\n
3.1.1

\n
\n
minor transient cleanup
\n
update .pot file
\n
fix to get all themes under both single and multisite installs
\n
\n
3.1.0

\n
\n
woot!! - updating from Bitbucket private repos now works!!
\n
fix to only add HTTP Authentication header under correct circumstances. This obviates need to fix for other APIs that might also use HTTP Authentication.
\n
fix to correctly add GitHub Access Token from $options to $download_link - oops
\n
changes $options to private static $options to save a few database calls
\n
Settings page only shows private repos, except for initial setup
\n
simpler test for checking branch as download endpoint
\n
correctly use parent:: instead of self::
\n
many updates for translation
\n
fix to ensure theme rollback and updating works in both single install and multisite
\n
fix to save settings from single site installations
\n
\n
3.0.7

\n
\n
more efficient solution to HTTP Authentication issues
\n
more efficient options cleanup
\n
remove some unnecessary code resulting in few database calls
\n
change default option setting to use add_site_option so not autoloading options
\n
\n
3.0.6

\n
\n
fix for other APIs that use HTTP Authentication, like JetPack - thanks @tsquez
\n
\n
3.0.5

\n
\n
fix more PHP Notices
\n
correctly set defaults for Settings page :P
\n
remove options for plugins/themes that are no longer present
\n
\n
3.0.4

\n
\n
Who would've thought file_exists was case-sensitive
\n
when checking meta, use empty() instead of ! isset() for null array
\n
set defaults for Settings page
\n
fix a number of PHP Notices
\n
\n
3.0.3

\n
\n
Bugfix to properly authenticate on JetPack Stats page
\n
\n
3.0.2

\n
\n
simplify check and exit on Settings if no Bitbucket plugins/themes
\n
\n
3.0.1

\n
\n
Remove Bitbucket settings from page if no appropriate plugins or themes exist.
\n
\n
3.0.0

\n
\n
Settings Page for your GitHub Access Tokens
\n
added POT file and some more i18n fixes - thanks @grappler
\n
added Requires WP and Requires PHP headers to set minimum version requirements - for @GaryJ
\n
move update check to function to also check WP and PHP version requirements.
\n
unset any HTTP Authorization headers for GitHub API calls as this gives a 401 error. Rare potential bug if you have private Bitbucket repos.
\n
\n
2.9.0

\n
\n
move instantiation of class GitHub_Plugin_Updater and class GitHub_Theme_Updater into GitHub_Updater::init() and restrict to current_user_can( 'update_plugins' ) and current_user_can( 'update_themes' ) so that non-privileged users don't incur load time.
\n
now loading classes via spl_autoload_register
\n
switched to erusev/parsedown for rendering changelogs, faster and more light-weight.
\n
now parses remote file info to save only file headers to transient. Hopefully speeds up database retrieval of transient.
\n
added README link to GitHub Link plugin by @szepeviktor
\n
added mu-plugin option and instructions.
\n
above revisions mostly due to @szepeviktor prodding me. ;-)
\n
accept CHANGES.md or CHANGELOG.md for processing, for @GaryJ
\n
composer support added, thanks @hyyan
\n
\n
2.8.1

\n
\n
fix for WP Coding Guidelines
\n
added check for upgrade process instead of $_GET['action'] (props @SLv99)
\n
launch classes from GitHub_Updater::init() so can load in add_action( 'init', ... from __construct(). Hopefully this will solve issues with remote upgraders like iThemes Sync, ManageWP, InfiniteWP, and MainWP. Thanks @jazzsequence for testing. Thanks @SLv99 for bringing this to my attention.
\n
\n
2.8.0

\n
\n
refactor API classes and class GitHub_Updater to add extra headers from API class. This should allow for better abstraction. Just need to call GitHub_Updater_{repo}_API::add_headers() in class GitHub_Plugin_Updater and class GitHub_ Theme_Updater.
\n
remove @since tags
\n
move maybe_authenticate_http to class GitHub_Updater_Bitbucket_API as it's not used elsewhere
\n
use non-strict check for http response code (thanks @echav)
\n
\n
2.7.1

\n
\n
added early exit if no local CHANGES.md file exists. This should save an API call.
\n
pull update from WP.org if plugin hosted in WP.org and branch is master.
\n
\n
2.7.0

\n
\n
created functions for getting and setting transients
\n
added deletion of all transients if force-check is used
\n
removed GitHub Timeout and Bitbucket Timeout headers
\n
fix for wp_remote_retrieve_response_code check
\n
give Seth Carstens proper credit in README.md
\n
move function make_rating to class GitHub_Updater
\n
fix for plugin name in update detail view
\n
fix for Bitbucket repo with no branch tag
\n
set default timeout to 12 hours, same as WP.org
\n
fix for 3.9 setting theme update details to display:none;
\n
fix for error when installing themes from WP.org repo
\n
fix for incorrect plugin upgrade link in detail popup
\n
\n
2.6.3

\n
\n
quick error checking fix for wp_remote_get error to wordpress.org API - thanks @deckerweb
\n
\n
2.6.1

\n
\n
fixed CHANGES.md for GFM strike-through
\n
\n
2.6.0

\n
\n
added transient to plugins_api call
\n
better zeroing of variables in getting local theme data
\n
add error checking to loading of classes
\n
set default transient timeout to 4 hours
\n
added new header GitHub Timeout or Bitbucket Timeout to set individual plugin/theme transient timeout
\n
fixed for Bitbucket private repos
\n
abide by WP Coding Guidelines, esp. for braces
\n
more error checking for correct variable fetch
\n
added graceful exit if repo does not exist
\n
\n
2.5.0

\n
\n
added class GitHub_Updater_Bitbucket_API for Bitbucket hosted plugins and themes.
\n
improvements to efficiency by not loading when DOING_AJAX
\n
improvements to efficiency in use of transients
\n
\n
2.4.5

\n
\n
set PHP MarkdownExtra posts and comments markup to false props @MikeHansonMe
\n
remove WP plugin header from markdown.php
\n
\n
2.4.4

\n
\n
forgot to include markdown.php - damn
\n
\n
2.4.2

\n
\n
removed PHP Markdown Lib as it required PHP >= 5.3 and that's higher than required by WordPress core.
\n
\n
2.4.1

\n
\n
switched from PHP Markdown Classic to the new PHP Markdown Lib to prevent collisions with other plugins, like Markdown On Save/Improved that also load PHP Markdown or PHP MarkdownExtra.
\n
\n
2.4.0

\n
\n
fixed transient assignment for tags returning empty array.
\n
added transient for CHANGES.md to themes, should further cut down on API 403 errors.
\n
new feature: theme rollback to previous version thanks @scarstens
\n
changed update methodology to use most recent tag first. If not tagged update from default branch.
\n
\n
2.3.3

\n
\n
fixed download link to have correct base URI for Repository Contents API. Oops.
\n
\n
2.3.2

\n
\n
rewrite of GitHub_Update_GitHub_API::construct_download_link to download zipball and provide appropriate endpoint.
\n
\n
2.3.1

\n
\n
now saving transient and adding early return if API returns 404, this should speed up plugin when repo doesn't have CHANGES.md file and provide for early return in no tags have been created. If no tags have been created the API is still hit.
\n
\n
2.3.0

\n
\n
moved action hook to remove after_theme_row_$stylesheet to class GitHub_Theme_Updater
\n
added feature: if branch other than master is specified then tagged version will be ignored. This should make it much easier for beta testing to groups. See README.md
\n
converted class GitHub_Update_GitHub_API to extension of class GitHub_Updater
\n
combined description and changelog to show in theme detail view. Rough formatting. Multisite only.
\n
greatly simplified bug fix from 2.2.2, now using Themes API.
\n
\n
2.2.2

\n
\n
bug fix for removing update notice for WP.org repo themes. Oops.
\n
\n
2.2.1

\n
\n
minor code simplifications
\n
many thanks to @grappler for solving how to remove default after_theme_row_$stylesheet
\n
\n
2.2.0

\n
\n
moved check and load for markdown.php into only function that uses it.
\n
minor README updates
\n
added abort if this plugin called directly
\n
added additional data to update available screen in both plugins and themes - issue #8
\n
removed requirement for tags in theme updating
\n
removed extra line endings from remote_version
\n
added ratings function for creating star ratings based upon GitHub repo data.
\n
bring parts of class GitHub_Theme_Updater code on par with class GitHub_Plugin_Updater
\n
added 'ghu-' prefix to transients
\n
ripped out theme rollback code. Moved to it's own branch on GitHub.
\n
add custom after_theme_update_{$stylesheet} detail.
\n
\n
2.1.1

\n
\n
bug fix to return early from call to plugins_api if not getting plugin information. Fixes issue with Plugin Search.
\n
\n
2.1.0

\n
\n
simplify check for class Markdown_Parser
\n
refactor to pass class GitHub_Update_GitHub_API as class object. This should enable the creation of other class objects for Bitbucket, etc.
\n
fix for setting branch when API not responding
\n
fix for setting download link when API not responding
\n
redesigned filter for setting transient timeout, but still not working (pull requests welcome)
\n
\n
2.0.1

\n
\n
bug fix to not load markdown.php twice. Just in case it's loaded by some other plugin.
\n
\n
2.0.0

\n
\n
rearranged where I put GitHub Plugin URI header, etc. in README and in this plugin.
\n
minor spelling fixes
\n
renamed some functions for their hooks
\n
refactored class GitHub_Plugin_Updater and class GitHub_Theme_Updater to use stdClass objects
\n
further refactored base class GitHub_Updater to contain renaming code and create stdClass objects for data.
\n
added some ability to see changelog for GitHub hosted plugins.
\n
trying to follow WordPress Plugin Boilerplate, so renamed classes to includes
\n
refactored putting all remote api calls in new class GitHub_Plugin_Updater_API.
\n
Theme updating should now be able to have a specified branch.
\n
works on WordPress 3.8
\n
included Michel Fortin's PHP-Markdown for rendering CHANGES.md
\n
\n
1.8.1

\n
\n
added some variable declarations
\n
added early return in no GitHub sourced plugins or themes are identified
\n
\n
1.8.0

\n
\n
refactored to use base class GitHub_Updater and extending classes GitHub_Plugin_Updater and GitHub_Theme_Updater.
\n
\n
1.7.4

\n
\n
changed method of not overwriting extra headers to pass array.
\n
\n
1.7.3

\n
\n
change '...' to &#8230 in renaming notification
\n
fix to not overwrite extra headers of other plugins.
\n
\n
1.7.2

\n
\n
removed sorting option from scandir. Doesn't work with older versions of PHP < 5.4.0
\n
removed extraneous data from array in multisite_get_themes
\n
\n
1.7.1

\n
\n
updated the transient for themes
\n
replaced readdir with scandir for creating WP_Theme object in multisite
\n
\n
1.7.0

\n
\n
updated class-theme-updater.php to utilize WP_Theme class
\n
added method get_remote_tag to update plugins using tags or branch, depending upon which has greater version number.
\n
get_remote_tag uses transient to limit calls to API
\n
fix for wp_get_themes not working under plugin network activation on multisite installation. I recreated wp_get_themes by reading in the theme directory and adding the WP_Theme object of wp_get_theme( 'dir_in_themes_dir' ) to an array.
\n
\n
1.6.1

\n
\n
bug fix for undeclared variable $github_plugins
\n
\n
1.6.0

\n
\n
Added separate method to parse plugin repo info from header
\n
Shortened GitHub Plugin URI to only use owner/repo
\n
Shortened GitHub Theme URI to only use owner/repo
\n
\n
1.5.0

\n
\n
Lots of documentation and some bug fixes. Thanks @GaryJones
\n
Made version checking regex more compatible. Thanks @GaryJones
\n
Added ability to define branch to update.
\n
Refactored plugin/theme renaming code.
\n
Added GitHub Branch feature - Thanks @GaryJones
\n
Trying to comply with WP Coding Standards.
\n
Major thanks to @GaryJones for all the pull requests and generally improving this project.
\n
\n
1.4.3

\n
\n
Fixed a couple of non-fatal PHP errors. Thanks @jazzsequence
\n
\n
1.4.2

\n
\n
Cleaned up readme's markdown.
\n
\n
1.4.1

\n
\n
Fixed the README to more accurately reflect support for both plugins and themes.
\n
\n
1.4

\n
\n
Fix for rename functions to be more precise, otherwise might rename wp.org repo themes.
\n
\n
1.3

\n
\n
Simplify a couple of if statements.
\n
\n
1.2

\n
\n
Fix to ignore renaming for wp.org plugins
\n
\n
1.1

\n
\n
Sanity check for theme api uri
\n
\n
1.0

\n
\n
Serialized WP_Theme object to search for added GitHub header, lots of help from Seth. No more file_get_contents.
\n
Converted plugin class and added it to make joint plugin/theme updater.
\n
\n
0.2

\n
\n
Code cleanup.
\n
Limit file_get_contents to 2000 bytes.
\n
\n
0.1

\n
\n
Initial commit
\n
"
            },
            "languages": "https://github.com/afragen/github-updater-translations",
            "ci_job": false,
            "release_asset": false,
            "broken": false,
            "banners": {
                "high": null,
                "low": null
            },
            "icons": {
                "svg": "https://www.tvmatchen.nu/plugins/github-updater/assets/icon.svg"
            },
            "remote_version": "7.5.0",
            "newest_tag": "7.5.0",
            "download_link": "https://api.github.com/repos/afragen/github-updater/zipball/7.5.0",
            "tags": [
                "5.3.4",
                "5.4.0",
                "5.4.1",
                "5.5.0",
                "5.6.0",
                "5.6.1",
                "5.6.2",
                "6.0.0",
                "6.1.0",
                "6.1.1",
                "6.2.0",
                "6.2.1",
                "6.2.2",
                "6.3.0",
                "6.3.1",
                "6.3.2",
                "6.3.3",
                "6.3.4",
                "6.3.5",
                "7.0.0",
                "7.1.0",
                "7.2.0",
                "7.3.0",
                "7.3.1",
                "7.4.0",
                "7.4.1",
                "7.4.2",
                "7.4.3",
                "7.4.4",
                "7.5.0"
            ],
            "rollback": {
                "7.5.0": "https://api.github.com/repos/afragen/github-updater/zipball/7.5.0",
                "7.4.4": "https://api.github.com/repos/afragen/github-updater/zipball/7.4.4",
                "7.4.3": "https://api.github.com/repos/afragen/github-updater/zipball/7.4.3",
                "7.4.2": "https://api.github.com/repos/afragen/github-updater/zipball/7.4.2",
                "7.4.1": "https://api.github.com/repos/afragen/github-updater/zipball/7.4.1",
                "7.4.0": "https://api.github.com/repos/afragen/github-updater/zipball/7.4.0",
                "7.3.1": "https://api.github.com/repos/afragen/github-updater/zipball/7.3.1",
                "7.3.0": "https://api.github.com/repos/afragen/github-updater/zipball/7.3.0",
                "7.2.0": "https://api.github.com/repos/afragen/github-updater/zipball/7.2.0",
                "7.1.0": "https://api.github.com/repos/afragen/github-updater/zipball/7.1.0",
                "7.0.0": "https://api.github.com/repos/afragen/github-updater/zipball/7.0.0",
                "6.3.5": "https://api.github.com/repos/afragen/github-updater/zipball/6.3.5",
                "6.3.4": "https://api.github.com/repos/afragen/github-updater/zipball/6.3.4",
                "6.3.3": "https://api.github.com/repos/afragen/github-updater/zipball/6.3.3",
                "6.3.2": "https://api.github.com/repos/afragen/github-updater/zipball/6.3.2",
                "6.3.1": "https://api.github.com/repos/afragen/github-updater/zipball/6.3.1",
                "6.3.0": "https://api.github.com/repos/afragen/github-updater/zipball/6.3.0",
                "6.2.2": "https://api.github.com/repos/afragen/github-updater/zipball/6.2.2",
                "6.2.1": "https://api.github.com/repos/afragen/github-updater/zipball/6.2.1",
                "6.2.0": "https://api.github.com/repos/afragen/github-updater/zipball/6.2.0",
                "6.1.1": "https://api.github.com/repos/afragen/github-updater/zipball/6.1.1",
                "6.1.0": "https://api.github.com/repos/afragen/github-updater/zipball/6.1.0",
                "6.0.0": "https://api.github.com/repos/afragen/github-updater/zipball/6.0.0",
                "5.6.2": "https://api.github.com/repos/afragen/github-updater/zipball/5.6.2",
                "5.6.1": "https://api.github.com/repos/afragen/github-updater/zipball/5.6.1",
                "5.6.0": "https://api.github.com/repos/afragen/github-updater/zipball/5.6.0",
                "5.5.0": "https://api.github.com/repos/afragen/github-updater/zipball/5.5.0",
                "5.4.1": "https://api.github.com/repos/afragen/github-updater/zipball/5.4.1",
                "5.4.0": "https://api.github.com/repos/afragen/github-updater/zipball/5.4.0",
                "5.3.4": "https://api.github.com/repos/afragen/github-updater/zipball/5.3.4"
            },
            "branches": {
                "develop": "https://api.github.com/repos/afragen/github-updater/zipball/develop",
                "gitea": "https://api.github.com/repos/afragen/github-updater/zipball/gitea",
                "master": "https://api.github.com/repos/afragen/github-updater/zipball/master",
                "no-backtrace": "https://api.github.com/repos/afragen/github-updater/zipball/no-backtrace"
            },
            "requires": "4.6",
            "tested": "5.0",
            "donate_link": "http://thefragens.com/github-updater-donate",
            "contributors": [
                "afragen",
                "garyj",
                "scarstens",
                "limikael"
            ],
            "downloaded": 0,
            "last_updated": "2018-03-12T00:55:43Z",
            "rating": 100,
            "num_ratings": 1876,
            "transient": {
                "GitHub Languages": "https://github.com/afragen/github-updater-translations",
                "GitHub Plugin URI": "https://github.com/afragen/github-updater",
                "Requires PHP": "5.3",
                "Requires WP": "4.6",
                "Name": "GitHub Updater",
                "PluginURI": "https://github.com/afragen/github-updater",
                "Version": "7.5.0",
                "Description": "A plugin to automatically update GitHub, Bitbucket, or GitLab hosted plugins, themes, and language packs. It also allows for remote installation of plugins or themes into WordPress.",
                "Author": "Andy Fragen",
                "TextDomain": "github-updater",
                "DomainPath": "/languages",
                "Network": "true",
                "dot_org": false
            },
            "repo_meta": {
                "private": false,
                "last_updated": "2018-03-12T00:55:43Z",
                "watchers": 1876,
                "forks": 293,
                "open_issues": 12
            },
            "watchers": 0,
            "forks": 0,
            "open_issues": 0,
            "requires_wp_version": "4.6",
            "requires_php_version": "5.3",
            "dot_org": false,
            "is_private": false,
            "language_packs": {
                "ar": {
                    "slug": "github-updater",
                    "language": "ar",
                    "updated": "2017-04-27 12:52-0700",
                    "package": "https://github.com/afragen/github-updater-translations/blob/master/packages/github-updater-ar.zip?raw=true",
                    "autoupdate": "1",
                    "type": "plugin",
                    "version": "7.5.0"
                },
                "bg_BG": {
                    "slug": "github-updater",
                    "language": "bg_BG",
                    "updated": "2017-04-27 13:32-0700",
                    "package": "https://github.com/afragen/github-updater-translations/blob/master/packages/github-updater-bg_BG.zip?raw=true",
                    "autoupdate": "1",
                    "type": "plugin",
                    "version": "7.5.0"
                },
                "de_AT": {
                    "slug": "github-updater",
                    "language": "de_AT",
                    "updated": "2017-04-27 12:44-0700",
                    "package": "https://github.com/afragen/github-updater-translations/blob/master/packages/github-updater-de_AT.zip?raw=true",
                    "autoupdate": "1",
                    "type": "plugin",
                    "version": "7.5.0"
                },
                "de_CH": {
                    "slug": "github-updater",
                    "language": "de_CH",
                    "updated": "2017-04-27 12:55-0700",
                    "package": "https://github.com/afragen/github-updater-translations/blob/master/packages/github-updater-de_CH.zip?raw=true",
                    "autoupdate": "1",
                    "type": "plugin",
                    "version": "7.5.0"
                },
                "de_DE": {
                    "slug": "github-updater",
                    "language": "de_DE",
                    "updated": "2017-04-27 13:33-0700",
                    "package": "https://github.com/afragen/github-updater-translations/blob/master/packages/github-updater-de_DE.zip?raw=true",
                    "autoupdate": "1",
                    "type": "plugin",
                    "version": "7.5.0"
                },
                "en_CA": {
                    "slug": "github-updater",
                    "language": "en_CA",
                    "updated": "2017-08-15 16:26-0700",
                    "package": "https://github.com/afragen/github-updater-translations/blob/master/packages/github-updater-en_CA.zip?raw=true",
                    "autoupdate": "1",
                    "type": "plugin",
                    "version": "7.5.0"
                },
                "es_ES": {
                    "slug": "github-updater",
                    "language": "es_ES",
                    "updated": "2017-04-27 13:39-0700",
                    "package": "https://github.com/afragen/github-updater-translations/blob/master/packages/github-updater-es_ES.zip?raw=true",
                    "autoupdate": "1",
                    "type": "plugin",
                    "version": "7.5.0"
                },
                "fr_FR": {
                    "slug": "github-updater",
                    "language": "fr_FR",
                    "updated": "2017-07-25 22:20+0200",
                    "package": "https://github.com/afragen/github-updater-translations/blob/master/packages/github-updater-fr_FR.zip?raw=true",
                    "autoupdate": "1",
                    "type": "plugin",
                    "version": "7.5.0"
                },
                "it": {
                    "slug": "github-updater",
                    "language": "it",
                    "updated": "2017-04-27 13:45-0700",
                    "package": "https://github.com/afragen/github-updater-translations/blob/master/packages/github-updater-it.zip?raw=true",
                    "autoupdate": "1",
                    "type": "plugin",
                    "version": "7.5.0"
                },
                "ja": {
                    "slug": "github-updater",
                    "language": "ja",
                    "updated": "2017-04-27 13:05-0700",
                    "package": "https://github.com/afragen/github-updater-translations/blob/master/packages/github-updater-ja.zip?raw=true",
                    "autoupdate": "1",
                    "type": "plugin",
                    "version": "7.5.0"
                },
                "pt_BR": {
                    "slug": "github-updater",
                    "language": "pt_BR",
                    "updated": "2017-04-27 13:01-0700",
                    "package": "https://github.com/afragen/github-updater-translations/blob/master/packages/github-updater-pt_BR.zip?raw=true",
                    "autoupdate": "1",
                    "type": "plugin",
                    "version": "7.5.0"
                },
                "pt_PT": {
                    "slug": "github-updater",
                    "language": "pt_PT",
                    "updated": "2017-11-06 12:07+0000",
                    "package": "https://github.com/afragen/github-updater-translations/blob/master/packages/github-updater-pt_PT.zip?raw=true",
                    "autoupdate": "1",
                    "type": "plugin",
                    "version": "7.5.0"
                },
                "ro_RO": {
                    "slug": "github-updater",
                    "language": "ro_RO",
                    "updated": "2017-04-27 12:39-0700",
                    "package": "https://github.com/afragen/github-updater-translations/blob/master/packages/github-updater-ro_RO.zip?raw=true",
                    "autoupdate": "1",
                    "type": "plugin",
                    "version": "7.5.0"
                },
                "ru_RU": {
                    "slug": "github-updater",
                    "language": "ru_RU",
                    "updated": "2017-04-27 12:33-0700",
                    "package": "https://github.com/afragen/github-updater-translations/blob/master/packages/github-updater-ru_RU.zip?raw=true",
                    "autoupdate": "1",
                    "type": "plugin",
                    "version": "7.5.0"
                },
                "sv_SE": {
                    "slug": "github-updater",
                    "language": "sv_SE",
                    "updated": "2017-05-09 09:30+0200",
                    "package": "https://github.com/afragen/github-updater-translations/blob/master/packages/github-updater-sv_SE.zip?raw=true",
                    "autoupdate": "1",
                    "type": "plugin",
                    "version": "7.5.0"
                },
                "uk": {
                    "slug": "github-updater",
                    "language": "uk",
                    "updated": "2017-04-27 12:16-0700",
                    "package": "https://github.com/afragen/github-updater-translations/blob/master/packages/github-updater-uk.zip?raw=true",
                    "autoupdate": "1",
                    "type": "plugin",
                    "version": "7.5.0"
                },
                "zh_TW": {
                    "slug": "github-updater",
                    "language": "zh_TW",
                    "updated": "2017-06-14 07:27-0700",
                    "package": "https://github.com/afragen/github-updater-translations/blob/master/packages/github-updater-zh_TW.zip?raw=true",
                    "autoupdate": "1",
                    "type": "plugin",
                    "version": "7.5.0"
                }
            }
        },
        "WordPress-Simple-History": {
            "type": "github_plugin",
            "uri": "https://github.com/bonny/WordPress-Simple-History",
            "enterprise": null,
            "enterprise_api": null,
            "owner": "bonny",
            "repo": "WordPress-Simple-History",
            "branch": "master",
            "slug": "bonny-WordPress-Simple-History-d11ecf3/index.php",
            "local_path": "/mnt/persist/www/docroot/plugins/WordPress-Simple-History/",
            "extended_repo": "github-bonny-WordPress-Simple-History",
            "author": "P\u00e4r Thernstr\u00f6m",
            "name": "Simple History",
            "local_version": "2.20",
            "sections": {
                "description": "Plugin that logs various things that occur in WordPress and then presents those events in a very * nice GUI. Av P\u00e4r Thernstr\u00f6m."
            },
            "languages": null,
            "ci_job": false,
            "release_asset": false,
            "broken": false,
            "banners": {
                "high": null,
                "low": null
            },
            "icons": [],
            "remote_version": "2.20",
            "newest_tag": "2.20",
            "download_link": "https://api.github.com/repos/bonny/WordPress-Simple-History/zipball/2.20",
            "tags": [
                "2.3.1",
                "2.4",
                "2.5",
                "2.5.1",
                "2.5.2",
                "2.5.3",
                "2.5.4",
                "2.5.5",
                "2.6",
                "2.7",
                "2.7.1",
                "2.7.2",
                "2.7.3",
                "2.7.4",
                "2.7.5",
                "2.8",
                "2.9",
                "2.9.1",
                "2.10",
                "2.11",
                "2.12",
                "2.13",
                "2.14",
                "2.14.1",
                "2.15",
                "2.16",
                "2.17",
                "2.18",
                "2.19",
                "2.20"
            ],
            "rollback": {
                "2.9.1": "https://api.github.com/repos/bonny/WordPress-Simple-History/zipball/2.9.1",
                "2.9": "https://api.github.com/repos/bonny/WordPress-Simple-History/zipball/2.9",
                "2.8": "https://api.github.com/repos/bonny/WordPress-Simple-History/zipball/2.8",
                "2.7.5": "https://api.github.com/repos/bonny/WordPress-Simple-History/zipball/2.7.5",
                "2.7.4": "https://api.github.com/repos/bonny/WordPress-Simple-History/zipball/2.7.4",
                "2.7.3": "https://api.github.com/repos/bonny/WordPress-Simple-History/zipball/2.7.3",
                "2.7.2": "https://api.github.com/repos/bonny/WordPress-Simple-History/zipball/2.7.2",
                "2.7.1": "https://api.github.com/repos/bonny/WordPress-Simple-History/zipball/2.7.1",
                "2.7": "https://api.github.com/repos/bonny/WordPress-Simple-History/zipball/2.7",
                "2.6": "https://api.github.com/repos/bonny/WordPress-Simple-History/zipball/2.6",
                "2.5.5": "https://api.github.com/repos/bonny/WordPress-Simple-History/zipball/2.5.5",
                "2.5.4": "https://api.github.com/repos/bonny/WordPress-Simple-History/zipball/2.5.4",
                "2.5.3": "https://api.github.com/repos/bonny/WordPress-Simple-History/zipball/2.5.3",
                "2.5.2": "https://api.github.com/repos/bonny/WordPress-Simple-History/zipball/2.5.2",
                "2.5.1": "https://api.github.com/repos/bonny/WordPress-Simple-History/zipball/2.5.1",
                "2.5": "https://api.github.com/repos/bonny/WordPress-Simple-History/zipball/2.5",
                "2.4": "https://api.github.com/repos/bonny/WordPress-Simple-History/zipball/2.4",
                "2.3.1": "https://api.github.com/repos/bonny/WordPress-Simple-History/zipball/2.3.1",
                "2.20": "https://api.github.com/repos/bonny/WordPress-Simple-History/zipball/2.20",
                "2.19": "https://api.github.com/repos/bonny/WordPress-Simple-History/zipball/2.19",
                "2.18": "https://api.github.com/repos/bonny/WordPress-Simple-History/zipball/2.18",
                "2.17": "https://api.github.com/repos/bonny/WordPress-Simple-History/zipball/2.17",
                "2.16": "https://api.github.com/repos/bonny/WordPress-Simple-History/zipball/2.16",
                "2.15": "https://api.github.com/repos/bonny/WordPress-Simple-History/zipball/2.15",
                "2.14.1": "https://api.github.com/repos/bonny/WordPress-Simple-History/zipball/2.14.1",
                "2.14": "https://api.github.com/repos/bonny/WordPress-Simple-History/zipball/2.14",
                "2.13": "https://api.github.com/repos/bonny/WordPress-Simple-History/zipball/2.13",
                "2.12": "https://api.github.com/repos/bonny/WordPress-Simple-History/zipball/2.12",
                "2.11": "https://api.github.com/repos/bonny/WordPress-Simple-History/zipball/2.11",
                "2.10": "https://api.github.com/repos/bonny/WordPress-Simple-History/zipball/2.10"
            },
            "branches": {
                "develop": "https://api.github.com/repos/bonny/WordPress-Simple-History/zipball/develop",
                "feature/log-acf": "https://api.github.com/repos/bonny/WordPress-Simple-History/zipball/feature/log-acf",
                "gh-pages": "https://api.github.com/repos/bonny/WordPress-Simple-History/zipball/gh-pages",
                "master": "https://api.github.com/repos/bonny/WordPress-Simple-History/zipball/master"
            },
            "requires": null,
            "tested": null,
            "donate_link": null,
            "contributors": [],
            "downloaded": 0,
            "last_updated": "2018-02-02T14:17:08Z",
            "rating": 100,
            "num_ratings": 115,
            "transient": {
                "GitHub Plugin URI": "https://github.com/bonny/WordPress-Simple-History",
                "Name": "Simple History",
                "PluginURI": "http://simple-history.com",
                "Version": "2.20",
                "Description": "Plugin that logs various things that occur in WordPress and then presents those events in a very * nice GUI.",
                "Author": "P\u00e4r Thernstr\u00f6m",
                "AuthorURI": "http://simple-history.com/",
                "TextDomain": "simple-history",
                "DomainPath": "/languages",
                "dot_org": false
            },
            "repo_meta": {
                "private": false,
                "last_updated": "2018-02-02T14:17:08Z",
                "watchers": 115,
                "forks": 35,
                "open_issues": 20
            },
            "watchers": 0,
            "forks": 0,
            "open_issues": 0,
            "requires_wp_version": "4.6",
            "requires_php_version": "5.3",
            "dot_org": false,
            "is_private": false
        },
        "tvm-ads-manager": {
            "type": "bitbucket_plugin",
            "uri": "https://bitbucket.org/tvmatchen/tvm-ads-manager",
            "enterprise": null,
            "enterprise_api": null,
            "owner": "tvmatchen",
            "repo": "tvm-ads-manager",
            "branch": "master",
            "slug": "tvm-ads-manager/tvm-ads-manager.php",
            "local_path": "/mnt/persist/www/docroot/plugins/tvm-ads-manager/",
            "extended_repo": "bitbucket-tvmatchen-tvm-ads-manager",
            "author": "Johan Str\u00f6m 2016",
            "name": "TVM Ads Manager",
            "local_version": "1.3",
            "sections": {
                "description": "ads ads ads!! Av Johan Str\u00f6m 2016."
            },
            "languages": null,
            "ci_job": false,
            "release_asset": false,
            "broken": false,
            "banners": {
                "high": null,
                "low": null
            },
            "icons": [],
            "remote_version": "1.3",
            "newest_tag": "0.0.0",
            "download_link": "https://bitbucket.org/tvmatchen/tvm-ads-manager/get/master.zip",
            "tags": [],
            "rollback": [],
            "branches": {
                "master": "https://bitbucket.org/tvmatchen/tvm-ads-manager/get/master.zip"
            },
            "requires": null,
            "tested": null,
            "donate_link": null,
            "contributors": [],
            "downloaded": 0,
            "last_updated": "2018-01-24T14:50:37.979894+00:00",
            "rating": 0,
            "num_ratings": 0,
            "transient": {
                "Bitbucket Plugin URI": "https://bitbucket.org/tvmatchen/tvm-ads-manager",
                "Name": "TVM Ads Manager",
                "PluginURI": "https://bitbucket.org/tvmatchen/tvm-ads-manager",
                "Version": "1.3",
                "Description": "ads ads ads!!",
                "Author": "Johan Str\u00f6m 2016",
                "AuthorURI": "http://www.swedishboy.se/",
                "dot_org": false
            },
            "repo_meta": {
                "private": true,
                "last_updated": "2018-01-24T14:50:37.979894+00:00",
                "watchers": 0,
                "forks": 0,
                "open_issues": 0
            },
            "watchers": 0,
            "forks": 0,
            "open_issues": 0,
            "requires_wp_version": "4.6",
            "requires_php_version": "5.3",
            "dot_org": false,
            "is_private": true
        }
    }
]`

**Comment by swedishboy** [comment]
any tip for where i should look. i see a huge update to sitemeta with all plugins i got installed (even those in wp's repositories). is this really correct?

i also see a later update to sitemeta with my themes and plugins where it says 'response\";a:0'

**Comment by afragen** [comment]
That is a proper argument sent with the wp-cron event.

Not sure what you mean by update to sitemeta. GitHub Updater does store data in the wp-options table for single installs and in the wp-sitemeta table for multisite installs. These are the correct tables when `set_site_option()` is used. There is one option stored per repository and the data is refreshed every 12 hours. These table rows are all removed if the plugin is deleted and are deleted and re-created if the _Refresh Cache_ button is used.

GitHub Updater does not store data regarding plugins/themes that it is not updating. Everything in the database written by GitHub Updater is prefixed with `ghu-`.

The stored data is parsed from as many as 6 API calls.

**Comment by afragen** [comment]
Please reopen and let me know if there are further issues.
